beacon-chain/sync/subscriber_data_column_sidecar.go

```diff
@@ -52,3 +54,30 @@ func (s *Service) receiveDataColumnSidecar(ctx context.Context, sidecar blocks.V
 
 	return nil
 }
+
+// allDataColumnSubnets returns the data column subnets for which we need to find peers
+// but don't need to subscribe to. This is used to ensure we have peers available in all subnets
+// when we are serving validators. When a validator proposes a block, they need to publish data
+// columns on all subnets. This method returns a nil map when there is no validators custody
+// requirement.
+func (s *Service) allDataColumnSubnets(_ primitives.Slot) map[uint64]bool {
+	validatorsCustodyRequirement, err := s.validatorsCustodyRequirement()
+	if err != nil {
+		log.WithError(err).Debug("Could not retrieve validators custody requirement")
```

Comment:
Should we log as ERROR instead DEBUG here?
I mean, if `err != nil`, we may fail to broadcast data column sidecars when proposing a block built locally. I think it's serious enough to warn the user even if they are not running the node with debug logs.

```suggestion
		log.WithError(err).Error("Could not retrieve validators custody requirement")
```